activemodel/lib/active_model/attribute.rb

```diff
@@ -98,7 +98,7 @@ def with_type(type)
 
     def dup_or_share # :nodoc:
       if @type.mutable?
-        dup
+        deep_dup
```

Comment:
While I agree it might be more appropriate to dup deeply here, is this method defined somewhere relevant, or is this just `dup` with extra steps?

Reply:
Yeah, we probably don't need this change. @xodene and I thought there might have been a change because of the change from `deep_dup` to `dup` in https://github.com/rails/rails/pull/53921, but upon more testing there didn't seem to be a change and we probably forgot to back this out.

Reply:
👍 Will change back to just `dup` here